include/uio.h

```diff
@@ -21,6 +21,20 @@ typedef struct uio {
   vm_map_t *uio_vmspace; /* destination address space */
 } uio_t;
 
+/* Uses struct subtyping enabled by `-fplan9-extensions` flag described in:
+ * https://gcc.gnu.org/onlinedocs/gcc/Unnamed-Fields.html */
+typedef struct uio_single {
```

Comment:
I think adding this structure and using -plan9 extensions is pretty fancy, but it's not widely used and makes code less readable. This is pretty nice feature to be used, but it would be weird to see it only in one place and I don't think it gives us any real benefits, except that it brings confusion to the people unfamiliar with this.